### PR TITLE
GEODE-9184: Add Radish ZLEXCOUNT command

### PR DESCRIPTION
**Pull request by jdeppe-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java [hunk 1]
```diff
@@ -357,6 +357,27 @@ long zcount(SortedSetScoreRangeOptions rangeOptions) {
     return addLimitToRange(rangeOptions, withScores, minIndex, maxIndex);
   }
 
+  long zlexcount(SortedSetLexRangeOptions lexOptions) {
+    // Assume that all members have the same score. Behaviour is unspecified otherwise.
+    double score = scoreSet.get(0).score;
```
Comment (dschneider-pivotal) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java:362]: Before you call get(0) do you need to do a check on scoreSet to see if it is empty?

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java [hunk 2]
```diff
@@ -357,6 +357,27 @@ long zcount(SortedSetScoreRangeOptions rangeOptions) {
...
+
+    AbstractOrderedSetEntry minEntry = new MemberDummyOrderedSetEntry(lexOptions.getMinimum(),
```
Comment (dschneider-pivotal) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java:364]: consider extracting these two blocks into getMinIndex(lexOptions, score) and getMaxIndex(lexOptions, score). These new methods could also be used by zrangebylex

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/NullRedisSortedSet.java [hunk 3]
```diff
@@ -63,6 +63,11 @@ long zcount(SortedSetScoreRangeOptions rangeOptions) {
     return 0;
   }
 
+  @Override
```
Comment (ringles) [inline geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/NullRedisSortedSet.java:66]: Again, alphabetizing is apparently what we're doing here.

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java (hunk 2 above)
Comment (ringles) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java:364]: If I can get the ZREVRANGEBYSCORE PR pushed up, we can use the helper methods I added there.

**Review by ringles — CHANGES_REQUESTED** [review]
Alphabetizing, and possibly waiting for ZREVRANGEBYSCORE to be merged for helper methods.

File: geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/data/RedisSortedSetTest.java [hunk 4]
```diff
@@ -278,6 +279,84 @@ public void zrange_shouldAlsoReturnScores_whenWithScoresSpecified() {
...
+  }
+
+  @Test
+  public void zlexcount_shouldBeExclusiveWhenSpecified() {
+    RedisSortedSet sortedSet = createRedisSortedSet(
+        "1.1", "member1",
+        "1.1", "member2",
+        "1.1", "member3",
+        "1.1", "member4",
+        "1.1", "member5");
```
Comment (nonbinaryprogrammer) [inline geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/data/RedisSortedSetTest.java:302]: looks like you're passing in the same values in all these tests that you added. might be worth abstracting that out. I don't think that having that there is crucial for test readability

**Review by nonbinaryprogrammer — COMMENTED** [review]
looks good, just one potential test improvement

**Comment by nonbinaryprogrammer** [comment]
actually one other thing: you forgot to add a test in the hits/misses integration test

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java (hunk 1 above)
Comment (jdeppe-pivotal) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java:335]: Done

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java (hunk 2 above)
Comment (jdeppe-pivotal) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java]: @ringles Could you point me at the methods you're referring to?

File: geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/NullRedisSortedSet.java (hunk 3 above)
Comment (jdeppe-pivotal) [reply geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/NullRedisSortedSet.java:66]: Done

File: geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/data/RedisSortedSetTest.java (hunk 4 above)
Comment (jdeppe-pivotal) [reply geode-apis-compatible-with-redis/src/test/java/org/apache/geode/redis/internal/data/RedisSortedSetTest.java]: Done

**Review by DonalEvans — APPROVED** [review]

**Review by ringles — APPROVED** [review]

**Review by nonbinaryprogrammer — APPROVED** [review]

**Pull request merged**